pkgs/by-name/wr/wrangler/package.nix

```diff
@@ -0,0 +1,90 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+  fetchFromGitHub,
+  pnpm_9,
+  nodejs,
+  npmHooks,
+  writeScriptBin,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "wrangler";
+  version = "3.62.0";
+
+  src = fetchFromGitHub {
+    owner = "cloudflare";
+    repo = "workers-sdk";
+    rev = "wrangler@${finalAttrs.version}";
+    hash = "sha256-/4iIkvSn85fkRggmIha2kRlW0MEwvzy0ZAmIb8+LpZQ=";
+  };
+
+  nativeBuildInputs = [
+    nodejs
+    pnpm_9.configHook
+  ];
+
+  pnpmDeps = pnpm_9.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    hash = "sha256-aTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=";
```

Comment:
how do you determine these hash? i'm trying to understand if an update script is no longer needed

Reply:
well, i found that pnpm doesn't have a helper for determining this, so i did this.
```
nix-build -E "with import <nixpkgs> {}; callPackage ./package.nix {}" 2>&1 | grep "got: " | awk '{print $NF}')
```
which i'm sure will cause no issues.

Reply:
> how do you determine the pnpmDeps hash?

I use my usual approach of inserting a bogus hash and letting the tools tell me the correct one - I probably should use a more adult approach ;-)

I changed it to this (just changed the first letter to `X`):

```
    hash = "sha256-XTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=";
```

and got this output when I did a build:

```
error: hash mismatch in fixed-output derivation '/nix/store/l1qcljcgammj59kdx0ng7y0303798g7d-wrangler-pnpm-deps.drv':
         specified: sha256-XTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=
            got:    sha256-aTTaiGXm1WYwmy+ljUC9yO3qtvN20SA+24T83dWYrI0=
```